apps/api/src/service/applicationActionService.ts

```diff
@@ -22,22 +22,47 @@ import { and, eq } from 'drizzle-orm';
 import { type PostgresDb } from '@/db/index.js';
 import { applicationActions } from '@/db/schemas/applicationActions.js';
 import { applicationActionsQuery } from '@/service/utils.js';
-import { failure, success } from '@/utils/results.js';
+import { type AsyncResult, failure, success } from '@/utils/results.js';
 import {
 	ApplicationActions,
 	ApplicationActionValues,
 	ApplicationStates,
 	ApplicationStateValues,
 } from '@pcgl-daco/data-model/src/types.js';
-import { ApplicationData, type ApplicationActionsColumnName, type OrderBy } from './types.js';
+import {
+	type ApplicationActionData,
+	type ApplicationActionsColumnName,
+	type ApplicationData,
+	type OrderBy,
+} from './types.js';
+
+/**
+ * ApplicationActionService provides methods for ApplicationActions DB access
+ * @param db - Drizzle Postgres DB Instance
+ * @method addActionRecord: Template method for adding an Action record
+ * @method create: Add Create action
+ * @method close: Add Close action
+ * @method draftSubmit: Add Submit action from Draft
+ * @method dacApproved: Add Approve action from DacReview
+ * @method dacRejected: Add Rejected action from DacReview
+ * @method dacRevision: Add Revision action from DacReview
+ * @method dacSubmit: Add Submit action from DacReview
+ * @method edit: Add Edit action
+ * @method repRevision: Add Revision action from INSTITUTIONAL_REP_REVIEW
+ * @method repApproved: Add Approved from INSTITUTIONAL_REP_REVIEW
+ * @method revoke: Add Revoked action
+ * @method withdraw: Add Revoked action
+ * @method getActionById: Find a specific Action record
+ * @method listActions: Find multiple Actions related to a given User or Application
+ */
```

Comment:
I removed return types for brevity and clarity, let me know if you'd find anything else valuable here